### PR TITLE
GDPR test

### PR DESCRIPTION
**Pull request by chartjes**
This test is designed to verify that content put into Kinto by a user is correctly removed when their Firefox account is deleted

### CODE REVIEW

**Comment by leplatrem** [comment]
I can't review this before tomorrow, @glasserc are you available to give Chris a hand please?

File: config-tests/test_gpdr_functionality.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    # Add data to Notes (kintotp)
+    tp_client = Client(server_url=conf.get(env, 'tp_server_url'), auth=auth)
+    tp_existing_records = tp_client.get_records(collection='notes', bucket='default')
+    assert len(tp_existing_records) == 0
+    data = {"subject": "QA Test", "value": "This stuff should get deleted"}
+    tp_client.create_record(data=data, collection='notes', bucket='default')
+    tp_updated_records = tp_client.get_records(collection='notes', bucket='default')
+    assert len(tp_updated_records) == len(tp_existing_records) + 1
+
+    # Grab the alias of the default bucket and set the permissions to make sure anyone can read it
```
Comment (glasserc) [inline config-tests/test_gpdr_functionality.py]: This comment says "set permissions", but it doesn't?

Maybe verify that doing the GET *before* deleting the account works?

File: config-tests/test_gpdr_functionality.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    bucket_id = we_client.server_info()["user"]["bucket"]
+
+    # Delete FxA account
+    acct.clear()
+    fxaclient.destroy_account(email, passwd)
+
+    # Wait 5 minutes and then make sure the bucket doesn't exist
+    time.sleep(301)
+    resp = requests.get(conf.get(env, 'we_server_url') + '/buckets/{0}'.format(bucket_id))
```
Comment (glasserc) [inline config-tests/test_gpdr_functionality.py]: Is this supposed to do a similar check for TP?

File: config-tests/test_gpdr_functionality.py (hunk 1 above)
Comment (chartjes) [reply config-tests/test_gpdr_functionality.py]: Yeah, I set the permissions further up. I am getting a 401 when trying to look at that collection

File: config-tests/test_gpdr_functionality.py (hunk 2 above)
Comment (chartjes) [reply config-tests/test_gpdr_functionality.py]: I chose not to do the TP check

**Comment by chartjes** [comment]
The only remaining thing is what pytest marker to give this test -- what stack will this belong to?

File: config-tests/test_gpdr_functionality.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import configparser
```
Comment (leplatrem) [inline config-tests/test_gpdr_functionality.py]: filename is wrong `General Data Protection Regulation` not `General Protection Data Regulation` :smirk:

File: config-tests/test_gpdr_functionality.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    )
+    we_record_id = we_record['data']['id']
+    we_updated_records = we_client.get_records(collection=conf.get(env, 'qa_collection'), bucket='default')
+    assert len(we_updated_records) == len(we_existing_records) + 1
+
+    # Add data to Notes (kintotp)
+    tp_client = Client(server_url=conf.get(env, 'tp_server_url'), auth=auth)
+    tp_existing_records = tp_client.get_records(collection='notes', bucket='default')
+    assert len(tp_existing_records) == 0
+    data = {"subject": "QA Test", "value": "This stuff should get deleted"}
```
Comment (leplatrem) [inline config-tests/test_gpdr_functionality.py]: nit: notes are encrypted too https://medium.com/firefox-test-pilot/notes-sync-feature-is-now-available-with-end-to-end-encryption-a4cf47bbf9ae

File: config-tests/test_gpdr_functionality.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    we_bucket_id = we_client.server_info()["user"]["bucket"]
+    resp = requests.get(
+        conf.get(env, 'we_server_url') +
+        '/buckets/{0}/collections/qa_collection/records/{1}'.format(we_bucket_id, we_record_id))
+    assert resp.status_code == 200
+    tp_bucket_id = tp_client.server_info()["user"]["bucket"]
+    resp = requests.get(
+        conf.get(env, 'tp_server_url') +
+        '/buckets/{0}/collections/notes/records/{1}'.format(tp_bucket_id, tp_record_id))
+    assert resp.status_code == 200
```
Comment (leplatrem) [inline config-tests/test_gpdr_functionality.py]: nit: you could also have a:
```py
anon_we_client = Client(server_url=conf.get(env, 'we_server_url'))
anon_we_client.get_record(id=we_record_id, bucket_id=we_bucket_id, collection="qa_collection")
```

File: config-tests/test_gpdr_functionality.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    tp_updated_records = tp_client.get_records(collection='notes', bucket='default')
+    assert len(tp_updated_records) == len(tp_existing_records) + 1
+
+    # Get the aliases of the buckets we are putting data in and
+    # make sure that an unauthenticated user can see these records
+    # before we delete the account
+    we_bucket_id = we_client.server_info()["user"]["bucket"]
+    resp = requests.get(
+        conf.get(env, 'we_server_url') +
+        '/buckets/{0}/collections/qa_collection/records/{1}'.format(we_bucket_id, we_record_id))
```
Comment (leplatrem) [inline config-tests/test_gpdr_functionality.py]: `qa_collection` should come from env here too

**Pull request merged**